linkup-cli/src/status.rs

```diff
@@ -272,3 +268,65 @@ pub fn server_status(url: String, extra_headers: Option<HeaderMap>) -> ServerSta
         Err(_) => ServerStatus::Error,
     }
 }
+
+fn prepare_services_statuses<I>(
+    session_name: &str,
+    services: I,
+) -> (Vec<ServiceStatus>, Receiver<(String, ServerStatus)>)
+where
+    I: Iterator<Item = LocalService> + Clone,
+{
+    let services_statuses: Vec<ServiceStatus> = services
+        .clone()
+        .map(|service| {
+            let url = match service.current {
+                ServiceTarget::Local => service.local.clone(),
+                ServiceTarget::Remote => service.remote.clone(),
+            };
+
+            let priority = service_priority(&service);
+
+            ServiceStatus {
+                name: service.name.clone(),
+                location: url.to_string(),
+                component_kind: service.current.to_string(),
+                status: ServerStatus::Loading,
+                service,
+                priority,
+            }
+        })
+        .collect();
+
+    let (tx, rx) = std::sync::mpsc::channel();
+
+    for service in services {
+        let tx = tx.clone();
+        let service_clone = service.clone();
+        let session_name = session_name.to_string();
+
+        thread::spawn(move || {
+            let status = service_status(&service_clone, &session_name);
+
+            tx.send((service_clone.name.clone(), status))
+                .expect("Failed to send service status");
+        });
+    }
+
+    drop(tx);
+
+    (services_statuses, rx)
+}
+
+fn is_internal_service(service: &LocalService) -> bool {
+    service.name == "linkup_local_server"
+        || service.name == "linkup_remote_server"
+        || service.name == "tunnel"
+}
+
+fn service_priority(service: &LocalService) -> i8 {
+    if is_internal_service(service) {
+        -1
+    } else {
+        10
+    }
```

Comment:
Why -1 and 10 instead of like 1 & 2?

Reply:
No big specific reason! I think in my mind it was `-1` for the Linkup internal services for higher prio and `10` for all services at the moment to give us some padding between 0..10 if want to explicit order some of them differently.
But I guess we can go with 1 and 2 🤔 Or maybe an enum with High and Low..
Wdyt?

Reply:
Changed to 1 and 2 on 3e05d2c1cf0bbaebbe6ec17770250c201fb85beb